syft/frameworks/torch/tensors/interpreters/large_precision.py

```diff
@@ -0,0 +1,226 @@
+    large values by packing them in smaller ones.
+    Typically a user will require to enlarge a float number by fixing its precision
+        tensor.fix_prec()
+    These smaller values are of type `internal_type`. The large value is defined by `precision_fractional`.
+
+    By default operations are done with NumPy. This implies unpacking the representation and packing it again.
+
+    Check the tests to see how to play with the different parameters.
+    """
+
+    def __init__(
+        self,
+        owner=None,
+        id=None,
+        tags=None,
+        description=None,
+        base: int = 10,
+        precision_fractional=0,
+        internal_type=torch.int32,
+        verbose=False,
+    ):
+        """Initializes a LargePrecisionTensor.
+
+        Args:
+            owner (BaseWorker): An optional BaseWorker object to specify the worker on which
+                the tensor is located.
+            id (str or int): An optional string or integer id of the LargePrecisionTensor.
+            tags (list): list of tags for searching.
+            description (str): a description of this tensor.
+            base (int): The base that will be used to to calculate the precision.
+            precision_fractional (int): The precision required by the caller.
+            internal_type (dtype): The large tensor will be stored using tensor of this type.
+        """
+        super().__init__(id=id, owner=owner, tags=tags, description=description)
+        self.base = base
+        self.internal_type = internal_type
+        self.precision_fractional = precision_fractional
+        self.verbose = verbose
+
+    def _create_internal_representation(self):
+        """Decompose a tensor into an array of numbers that represent such tensor with the required precision"""
+        result = []
+        for x in np.nditer(self.child):
+            n = int(x.item() * self.base ** self.precision_fractional)
+            if self.verbose:
+                print(f"\nAdding number {n} for item {x.item()}\n")
+            result.append(
+                LargePrecisionTensor._split_number(n, self.internal_precision, self.internal_type)
+            )
+        max_length = len(max(result, key=len))
+        new_shape = self.child.shape + (max_length,)
+        # List with different lengths will result in np.array building an array of lists instead of the matrices
+        # itertools.zip_longest fills on the right but we need to fill with zeros on the left
+        n_result = []
+        for a_number in result:
+            len_a_number = len(a_number)
+            if len_a_number < max_length:
+                n_result.append([0] * (max_length - len_a_number) + a_number)
+            else:
+                n_result.append(a_number)
+        result = np.array(n_result).reshape(new_shape)
+        return torch.tensor(result, dtype=self.internal_type)
```

Comment:
Avoiding the loop over all the elements is not too hard, you can just change _split_number so that it takes an array instead of a single value (I did it but I'm not sure I can push to this PR, I can send you the small changes though)